xonsh/history.py

```diff
@@ -392,13 +431,15 @@ def _main(hist, args):
     if (args[0] == 'show' and len(args) > 1 and args[-1].startswith('-') and
             args[-1][1].isdigit()):
         args.insert(-1, '--')  # ensure parsing stops before a negative int
-    parser = _create_parser()
-    ns = parser.parse_args(args)
+    ns = _HIST_PARSER.parse_args(args)
     if ns.action is None:  # apply default action
-        ns = parser.parse_args(['show'] + args)
+        ns = _HIST_PARSER.parse_args(['show'] + args)
     _MAIN_ACTIONS[ns.action](ns, hist)
 
+
 def main(args=None, stdin=None):
     """This is the history command entry point."""
     _ = stdin
-    _main(builtins.__xonsh_history__, args)
+    _main(builtins.__xonsh_history__, args)  # pylint: disable=no-member
+
+_HIST_PARSER = _create_parser()
```

Comment:
Oh I see. You have moved this to be created when you import the module. I'm not sure that I agree with this. Doing it this way creates the parser when xonsh is started up. Doing it the way that I had originally only creates the parser if and when the `history` command is used. The way implemented here adds to the startup time, which we get a lot of complaints about and so I try to be as careful as possible.